plugins/modules/aci_vmm_enhanced_lag_policy.py

```diff
@@ -0,0 +1,340 @@
+            "lacpEnhancedLagPol": {
+                "attributes": {
+                    "name": "test_enhanced_lag_policy",
+                    "mode": "active",
+                    "lbmode": "src-dst-ip",
+                    "numLinks": "4",
+                    "dn": "uni/vmmp-VMware/dom-test_vmm_dom/vswitchpolcont/enlacplagp-test_enhanced_lag_policy"
+                }
+            }
+        }
+    ]
+error:
+  description: The error information as returned from the APIC.
+  returned: failure
+  type: dict
+  sample:
+    {
+        "code": "801",
+        "text": "property name of enlacplagp-test_enhanced_lag_policy failed validation"
+    }
+proposed:
+  description: The configuration sent to the APIC.
+  returned: info
+  type: dict
+  sample:
+    {
+        "lacpEnhancedLagPol": {
+            "attributes": {
+                "name": "test_enhanced_lag_policy",
+                "mode": "active",
+                "lbmode": "src-dst-ip",
+                "numLinks": "4"
+            }
+        }
+    }
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.aci.plugins.module_utils.aci import (
+    ACIModule,
+    aci_argument_spec,
+    enhanced_lag_spec,
+)
+from ansible_collections.cisco.aci.plugins.module_utils.aci import (
+    aci_annotation_spec,
+    aci_owner_spec,
+)
+
+VM_PROVIDER_MAPPING = dict(
+    cloudfoundry="CloudFoundry",
+    kubernetes="Kubernetes",
+    microsoft="Microsoft",
+    openshift="OpenShift",
+    openstack="OpenStack",
+    redhat="Redhat",
+    vmware="VMware",
+)
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(aci_annotation_spec())
+    argument_spec.update(aci_owner_spec())
+    argument_spec.update(enhanced_lag_spec(name_is_required=False))
+    argument_spec.update(
+        domain=dict(type="str", aliases=["domain_name", "domain_profile"]),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+        vm_provider=dict(type="str", choices=list(VM_PROVIDER_MAPPING.keys())),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["name", "domain", "vm_provider"]],
+            ["state", "present", ["name", "domain", "vm_provider"]],
+        ],
+    )
+
+    name = module.params.get("name")
+    lacp_mode = module.params.get("lacp_mode")
+    load_balancing_mode = module.params.get("load_balancing_mode")
+    number_uplinks = module.params.get("number_uplinks")
+    domain = module.params.get("domain")
+    state = module.params.get("state")
+    vm_provider = module.params.get("vm_provider")
+
+    aci = ACIModule(module)
+
+    aci.construct_url(
+        root_class=dict(
+            aci_class="vmmProvP",
+            aci_rn="vmmp-{0}".format(VM_PROVIDER_MAPPING.get(vm_provider)),
+            module_object=vm_provider,
+            target_filter={"name": vm_provider},
```

Comment:
I think vmmProvP idtenifier is`vendor`
```suggestion
            target_filter={"vendor": vm_provider},
```